src/contract.ts

```diff
@@ -45,14 +44,42 @@ export class Contract {
             Object.defineProperty(this, methodName, {
                 writable: false,
                 enumerable: true,
-                value: nameFunction(methodName, async (args: object = {}, gas?: BN, amount?: BN, ...ignored) => {
-                    if (ignored.length || !(isObject(args) || isUint8Array(args))) {
-                        throw new PositionalArgsError();
+                value: nameFunction(
+                    methodName,
+                    async (
+                        args: object = {},
+                        optionsOrGas?: ChangeMethodOptions | BN,
+                        deposit?: BN,
+                        ...ignored
+                    ) => {
+                        const options = isObject(optionsOrGas)
+                            ? optionsOrGas as ChangeMethodOptions
+                            : {
+                                gas: (() => {
```

Comment:
this construction seems a bit complicated. Why not just have `if` at top-level?

Reply:
Updated to:

```ts
let options = optionsOrGas as ChangeMethodOptions | undefined;
if (!options && !deposit) {
    options = {} as ChangeMethodOptions;
} else if (!options || !(options.gas || options.deposit || options.meta)) {
    // passed positional gas or deposit
    const deprecate = depd('positional gas & deposit amount');
    deprecate('pass { gas: inUnits, deposit: inYoctoNEAR } instead');
    options = {
        gas: optionsOrGas as BN,
        deposit
    } as ChangeMethodOptions;
}
```